### ISSUE TITLE
Some TCP-specific code leaked into twisted/internet/test/connectionmixins.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5578]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5578|
|Type|defect|
|Created|2012-03-28 12:51:58Z|
|Branch|https://github.com/twisted/twisted/tree/stream-client-tests-5578-2|

twisted/internet/test/connectionmixins.py is intended to hold code useful for any connection-oriented ports/transports.  The tests it defines should _all_ be applicable to any kind of SOCK_STREAM sockets.  In particular, that means they should run on all of TCP, SSL, and UNIX.

`TCPClientTestsMixin` is limited to TCP (as you might expect from the name).  It needs to be generalized to apply to TCP, SSL, and UNIX.

<details><summary>Searchable metadata</summary>

```
trac-id__5578 5578
type__defect defect
reporter__exarkun exarkun
priority__high high
milestone__ 
branch__branches_stream_client_tests_5578_2 branches/stream-client-tests-5578-2
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__tests tests
time__1332939118000000 1332939118000000
changetime__1363624679000000 1363624679000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)|@therve set owner to @therve|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve commented|
|-|-|

(In [34977]) Branching to 'stream-client-tests-5578'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve commented|
|-|-|

(In [37460]) Branching to 'stream-client-tests-5578-2'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve removed owner|
|-|-|

Alright, it looks TCP-free. I used the new mixin in Unix tests, not in SSL yet. I think we can do that in a next branch if that's OK.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_5"></a>|@tomprince set owner to @therve|
|-|-|

1.test_addresses
  - I wouldn't wrap the `L{}` even if it goes over the limit. 
  - You are also losing the verification of the `repr`.
 2. UnixClientTestsBuilder
  - This should have a test of the `repr`, as well.
 3. There are a [number](https://buildbot.twistedmatrix.com/builders/twistedchecker/builds/342) of twistedchecker failures. But I'm not sure if any of them are worth fixing. I think perhaps putting listen/connect (raising `NotImplementedError`) in the base class, and putting docs (with epytext markup) might be the right thing for some of them.

Please merge after dealing with `repr` and optionally the other points.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set status to `closed`|
|-|-|

(In [37572]) Merge stream-client-tests-5578-2

Author: therve
Reviewer: tom.prince
Fixes: [#5578](https://github.com/twisted/twisted/issues/5578)

Remove TCP specific bits in the connectionmixins test module, and reuse it in
test_unix.
